File: leetcode/1441.用栈操作构建数组.py
#
# @lc app=leetcode.cn id=1441 lang=python3
#
# [1441] 用栈操作构建数组
#


from typing import List


# @lc code=start
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        res = []
        pre = 0
        for i in target:
            if i - pre > 1:
                res.extend(
                    ["Push", "Pop"] * (i - pre - 1)
                )
            res.append("Push")
            pre = i
        return res


# @lc code=end
print(Solution().buildArray(target=[1, 3], n=3))
print(Solution().buildArray(target=[1, 2, 3], n=3))
print(Solution().buildArray(target=[2, 3, 4], n=4))

File: leetcode/1441.用栈操作构建数组.cpp
#include <iostream>
#include <vector>
using namespace std;
/*
 * @lc app=leetcode.cn id=1441 lang=cpp
 *
 * [1441] 用栈操作构建数组
 */

// @lc code=start
class Solution
{
public:
    vector<string> buildArray(vector<int> &target, int n)
    {
        vector<string> res;
        int pre = 0;
        for (auto &&i : target)
        {
            if (i - pre > 1)
            {
                for (size_t j = 0; j < i - pre - 1; j++)
                {
                    res.push_back("Push");
                    res.push_back("Pop");
                }
            }
            res.push_back("Push");
            pre = i;
        }
        return res;
    }
};
// @lc code=end
